.github/workflows/workflow.yml

```diff
@@ -0,0 +1,29 @@
+name: Run unit tests
+on: ["push", "pull_request"]
+
+jobs:
+  test:
+    name: Dub tests
+    strategy:
+      matrix:
+        os: [macOS-latest, ubuntu-latest, windows-latest]
+        dc: [dmd-latest, ldc-latest]
+        exclude:
+          - { os: macOS-latest, dc: dmd-latest }
+
+    runs-on: ${{ matrix.os }}
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: 'Install D compiler'
+      uses: dlang-community/setup-dlang@v1
+      with:
+        compiler: ${{ matrix.dc }}
+
+    - name: 'Build and test'
+      run: |
+        # Test in release mode
+          dub test --build=release
+          # Coverage
+          dub test --build=cov
```

Comment:
This was a problem with a dub that I prefer to circumvent in another way. To run coverage in unittest mode you can create a configuration for that and override the build modes. I would also suggest to add build modes to the matrix to test in `debug`, `release`, `release-debug`,`release-nobounds` and create another build mode to also force inlining `release-inline`. This will cover much more compilation flags and avoid weird bugs produced by the DMD backend by testing against them.

Reply:
How can I do that?

Reply:
```json
  "buildTypes": {
    "debug": {
      "buildOptions": [ "debugMode", "debugInfo" ]
    },
    "release": {
      "buildOptions": [ "releaseMode", "optimize", "noBoundsCheck" ]
    },
    "release-debug": {
      "buildOptions": [ "releaseMode", "optimize", "noBoundsCheck", "debugInfo" ]
    },
    "release-inline": {
      "buildOptions": [ "releaseMode", "optimize", "inline", "noBoundsCheck" ]
    }
  }
```

Testing with those build types should be enough.

Dub already have a unittest default configuration, AFAIK, you should be able to run it with `--config=unittest` and add coverage separately using `--coverage` argument.